### ISSUE TITLE
AboutSlint component causing ESP32S3 restart on T-Display-S3 board

### ISSUE THREAD

**Issue by Bulion** [slint-ui/slint#3439]
Hello team,

TL;DR
At T-Display-S3 board while using C++ backend AboutSlint{} is causing fall into neverending for loop which reset of the board (without any error).

I've started today my small side project which I want to develop on T-Display-S3 board from LILYGO using new release of Slint 1.2 which supports C++ backend on MCU. While creating a display and touch initialization code I was facing a strange issue which was causing board to reset after executing `ui->run();` function. At first point I was thinking that it was my fault connected to wrong configuration of LCD but after confirming that I'm able to draw on screen without use of Slint I've started debugging its internal components. This brings me to the file `software_renderer.rs` where I've find out that for loop going over components at line 256 is executing forever. This directed me to try changing something in .slint file which was your basic example containing AboutSlint{} and Text component. After removing AboutSlint{} suddenly it started to work. I didn't check how this component is actually made but maybe because screen is only 170px wide it might cause the issue? Anyway if you would like to test that itself just use my repo and add AboutSlint{} to .slint file: https://github.com/OakDevices/ESP32-T-Display-S3-Slint-Cpp-Template

Also if you would like me to test something as I have board on my desk just ask :).

I'm using:
- Slint 1.2
- Rustc 1.72.0
- ESP-IDF 5.1.1
- Builtin driver for ST7789 display controller

P.S.: Code snippets at https://components.espressif.com/components/slint/slint contains some small errors:
- Some characters like `&` or `>` are replaced by `&gt;` and `&amp;`
- `slint::platform::rgb565pixel` should be replaced by `slint::platform::Rgb565Pixel`
- Also two includes does not look right: `#include <bsp display.h="">` and `#include <bsp esp-bsp.h="">` but maybe it is some extension in xtensa compiler that I'm not aware?

BR,
Dawid

**Comment by tronical** [comment]
Thanks for trying out Slint on ESP-IDF and the report :)

Unfortunately I don't have the T-Display-S3 board (although now that I see it, I want one - it looks really cool! ;-). But I've seen this symptom before. In my experience there are two possible reasons:

1. You're running out of stack space. Make sure to allocate maybe +8K in `idf.py menuconfig` under `Component config --> ESP System Settings --> Main task stack size`.
2. You're running out of RAM. Your device appears to be equipped with 8MB PSRAM. Make sure to [enable PSRAM](https://docs.espressif.com/projects/esp-idf/en/latest/esp32s3/api-guides/flash_psram_config.html#configure-the-psram) that in your `idf.py menuconfig` as well. I'm not sure about this specific board. It's possible that you can use the same settings that we use for the [S3 Box](https://github.com/slint-ui/slint/blob/master/examples/carousel/esp-idf/s3-box/sdkconfig.defaults).

If those don't help, could you perhaps paste the output of `idf.py monitor` here when the controller resets itself? Perhaps there's something in the output that helps further.

Regarding the code snippets: Yes, this is an issue in the esp-idf components registry. I've reported it upstream last week at  https://github.com/espressif/idf-component-manager/issues/42 . Apologies for the compile errors :)

**Comment by Bulion** [comment]
Hi,

Ad. 1 I've adjusted it to +8K beforehand and that was not the issue. 
Ad. 2 In fact after enabling PSRAM it was detected by MCU and now I'm able to display AboutSlint screen. It is quite weird as basing on the official board schematic I cannot see PSRAM chip connected to MCU and the docs states there is 8MB one :D.

In fact there a new issue emerge after enabling PSRAM. When I press the touch button (trigger render loop) the color lines appears at the right side of the screen like on attached picture. 
![DSC_2127](https://github.com/slint-ui/slint/assets/6791070/4e469bad-e683-4ec7-9bdb-b5695b77f225)

Did you face such issue? I was already trying to adjust data alignment for the display but it did not help.

**Comment by tronical** [comment]
That's odd - I've never seen such an issue before. Is this visual artefact "stable", or is there any kind of flickering involved?

@ogoffart Does this ring any bell for you?

**Comment by Bulion** [comment]
It is flickering a bit while I'm pressing the button. At startup the screen is without glitch. Only after pressing button (rerendering color of rectangle) is causing this lines to show up. Also I've tried without AboutSlint component and it behaves the same.

**Comment by ogoffart** [comment]
I have not seen that before.

**Comment by ogoffart** [comment]
Could you try again with Slint 1.3? 
commit 22284000ecc0c5fc37d45a45c497a4c527f95645  added a vsync wait may fix the artifacts.

**Comment by isthaison** [comment]
![image](https://github.com/slint-ui/slint/assets/22739578/119fe581-d179-48f6-8ec2-526df877b75f)
 With lasted version, I also seen this case

**Comment by ogoffart** [comment]
This is a strange pattern. I don't explain it.  It looks like maybe some kind of corruption happens when sending pixel to the screen. But we just render to a buffer and delegate that to the underlying esp library. What hardware is this exactly?

**Comment by isthaison** [comment]
https://www.lilygo.cc/products/t-display-s3?variant=42589373268149,  This is board lilygo  T-Display-S3, version touch with driver cst816

**Comment by isthaison** [comment]
https://github.com/slint-ui/slint/assets/22739578/6eee56f3-5a9f-46a9-9e4d-7995f37eè058
I think this case when pixel rollback color

**Comment by ogoffart** [comment]
Unfortunately, i can't reproduce that to any of the device i have, so i'm not sure how i can go forward with this issue

**Comment by ogoffart** [comment]
Maybe 875ca075fb5b2dfe4c3ab0a499d5759412fc1395 actually fixed that.
Does the problem still exist with 1.9?

**Comment by ogoffart** [comment]
Closing an old issue. Please comment if the problem still exist.
